cmd/dra-example-kubeletplugin/state.go

```diff
@@ -173,3 +230,78 @@ func (s *DeviceState) prepareDevices(claim *resourceapi.ResourceClaim) (Prepared
 func (s *DeviceState) unprepareDevices(claimUID string, devices PreparedDevices) error {
 	return nil
 }
+
+func (p *PreparedDevice) applyConfig() error {
+	return nil
+}
+
+// GetDevices extracts the list of drapbv1.Devices from PreparedDevices.
+func (pds PreparedDevices) GetDevices() []*drapbv1.Device {
+	var devices []*drapbv1.Device
+	for _, pd := range pds {
+		devices = append(devices, &pd.Device)
+	}
+	return devices
+}
+
+// GetOpaqueDeviceConfigs returns an ordered list of configs specified for a device request in a resource claim.
+//
+// Configs can either come from the resource claim itself or from the device
+// class associated with the request. Configs coming directly from the resource
+// claim take precedence over configs coming from the device class. Moreover,
+// configs found later in the list of configs attached to its source take
+// precedence over configs found earlier in the list for that source.
+//
+// All of the configs relevant to the specified request for this driver will be
+// returned in order of precedence (from lowest to highest). If no config is
+// found, nil is returned.
+func GetOpaqueDeviceConfigs(
+	decoder runtime.Decoder,
+	driverName,
+	request string,
+	possibleConfigs []resourceapi.DeviceAllocationConfiguration,
+) ([]runtime.Object, error) {
+	// Collect all configs in order of reverse precedence.
+	var classConfigs []resourceapi.DeviceConfiguration
+	var claimConfigs []resourceapi.DeviceConfiguration
+	var candidateConfigs []resourceapi.DeviceConfiguration
+	for _, config := range possibleConfigs {
+		if !(len(config.Requests) == 0 || slices.Contains(config.Requests, request)) {
+			continue
+		}
+		switch config.Source {
+		case resourceapi.AllocationConfigSourceClass:
+			classConfigs = append(classConfigs, config.DeviceConfiguration)
+		case resourceapi.AllocationConfigSourceClaim:
+			claimConfigs = append(claimConfigs, config.DeviceConfiguration)
+		default:
+			return nil, fmt.Errorf("invalid config source: %v", config.Source)
+		}
+	}
+	candidateConfigs = append(candidateConfigs, classConfigs...)
+	candidateConfigs = append(candidateConfigs, claimConfigs...)
+
+	// Walk all configs from the candidateConfigs slice in the order they were added.
+	// The latest config matching the GroupVersionKind of the defaultConfig
+	// will be returned. If no configs match, then the original defaultConfig
+	// is returned.
```

Comment:
```suggestion
	// Decode all configs that are relevant for the driver.
```

I think this comment is mostly out-dated.

Reply:
Updated.